Cannot build on OSX

orsamac:avian orsa$ make
compiling build/darwin-x86_64/java-io.o
classpath/java-io.cpp:18:17: error: jni.h: No such file or directory
In file included from classpath/java-io.cpp:19:
classpath/jni-util.h:79: error: variable or field ‘throwNew’ declared void
classpath/jni-util.h:79: error: ‘JNIEnv’ was not declared in this scope
classpath/jni-util.h:79: error: ‘e’ was not declared in this scope
classpath/jni-util.h:79: error: expected primary-expression before ‘const’
classpath/jni-util.h:79: error: expected primary-expression before ‘const’
classpath/jni-util.h:79: error: expected primary-expression before ‘...’ token
In file included from /usr/include/dirent.h:65,
                 from classpath/java-io.cpp:71:
/usr/include/sys/dirent.h:86: error: expected declaration before end of line
make: **\* [build/darwin-x86_64/java-io.o] Error 1
orsamac:avian orsa$

--- a/README.md
+++ b/README.md
@@ -6,35 +6,37 @@ Avian - A lightweight Java Virtual Machine (JVM)
 Quick Start
 -----------
 
+These are examples of building Avian on various operating systems for
+the x86_64 architecture.  You may need to modify JAVA_HOME according
+to where the JDK is installed on your system.  In all cases, be sure
+to use forward slashes in the path.
+
 #### on Linux:
-    $ export JAVA_HOME=/usr/local/java # or wherever you have the JDK installed
+    $ export JAVA_HOME=/usr/lib/jvm/java-7-openjdk-amd64
     $ make
-    $ build/linux-i386/avian -cp build/linux-i386/test Hello
+    $ build/linux-x86_64/avian -cp build/linux-x86_64/test Hello
 
 #### on Mac OS X:
-    $ export JAVA_HOME=/Library/Java/Home
+    $ export JAVA_HOME=$(/usr/libexec/java_home)
     $ make
-    $ build/darwin-i386/avian -cp build/darwin-i386/test Hello
+    $ build/darwin-x86_64/avian -cp build/darwin-x86_64/test Hello
  
 #### on Windows (MSYS):
-    $ git clone git@github.com:ReadyTalk/win32.git ../win32
-    $ export JAVA_HOME="C:/Program Files/Java/jdk1.6.0_07"
+    $ git clone git@github.com:ReadyTalk/win64.git ../win64
+    $ export JAVA_HOME="C:/Program Files/Java/jdk1.7.0_45"
     $ make
-    $ build/windows-i386/avian -cp build/windows-i386/test Hello
+    $ build/windows-x86_64/avian -cp build/windows-x86_64/test Hello
 
 #### on Windows (Cygwin):
-    $ git clone git@github.com:ReadyTalk/win32.git ../win32
-    $ export JAVA_HOME="/cygdrive/c/Program Files/Java/jdk1.6.0_07"
+    $ git clone git@github.com:ReadyTalk/win64.git ../win64
+    $ export JAVA_HOME="/cygdrive/c/Program Files/Java/jdk1.7.0_45"
     $ make
-    $ build/windows-i386/avian -cp build/windows-i386/test Hello
+    $ build/windows-x86_64/avian -cp build/windows-x86_64/test Hello
 
 #### on FreeBSD:
-    $ export JAVA_HOME=/usr/local/openjdk7 # or wherever you have the JDK installed
+    $ export JAVA_HOME=/usr/local/openjdk7
     $ gmake
     $ build/freebsd-x86_64/avian -cp build/freebsd-x86_64/test Hello
-
-Adjust JAVA_HOME according to your system, but be sure to use forward
-slashes in the path.
 
 
 Introduction

